test/facade_converter_test.ts

```diff
@@ -237,6 +248,52 @@ var y = x.length;`);
 }`);
   });
 
+  describe('promises', () => {
+    it('translates into Futures', () => {
+      expectWithTypes('let x: Promise = Promise.resolve(1);').to.equal(`import "dart:async";
+
+Future x = Future.value(1);`);
+      expectWithTypes('let x: Promise = Promise.reject(1);').to.equal(`import "dart:async";
+
+Future x = Future.error(1);`);
+      expectWithTypes('let x: Promise = new Promise((resolve) => {resolve(1);});')
+          .to.equal(`import "dart:async";
+
+Future x = (() {
```

Comment:
The Completer object is the closest to something that works. This trick is a lot more code but keeps the interface intact. I can move it somewhere else but that would require the client to have a dependency to a library. It would also assume we need to emit code in the global scope... which I think we don't support right now.